src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/StacksTests.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using Xunit.Abstractions;
+using Xunit;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.Runners;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Runners;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.HttpApi;
+using Microsoft.Extensions.DependencyInjection;
+using System.IO;
+using System.Text.Json;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    [Collection(DefaultCollectionFixture.Name)]
+    public class StackTests
+    {
+#if NET6_0_OR_GREATER
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ITestOutputHelper _outputHelper;
+
+        private const string ExpectedModule = @"Microsoft.Diagnostics.Monitoring.UnitTestApp.dll";
+        private const string ExpectedClass = @"Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios.StacksWorker+StacksWorkerNested`1[System.Int32]";
+        private const string ExpectedFunction = @"DoWork[System.Int64]";
+        private const string ExpectedCallbackFunction = @"Callback";
+        private const string NativeFrame = "[NativeFrame]";
+
+        public StackTests(ITestOutputHelper outputHelper, ServiceProviderFixture serviceProviderFixture)
+        {
+            _httpClientFactory = serviceProviderFixture.ServiceProvider.GetService<IHttpClientFactory>();
+            _outputHelper = outputHelper;
+        }
+
+        [Fact]
+        public Task TestPlainTextStacks()
+        {
+            return ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                WebApi.DiagnosticPortConnectionMode.Listen,
+                TestAppScenarios.Stacks.Name,
+                appValidate: async (runner, client) =>
+                {
+                    int processId = await runner.ProcessIdTask;
+
+                    using ResponseStreamHolder holder = await client.CaptureStacksAsync(processId, plainText: true);
+                    Assert.NotNull(holder);
+
+                    using StreamReader reader = new StreamReader(holder.Stream);
+                    string line = null;
+
+                    string[] expectedFrames =
+                    {
+                        FormatFrame(ExpectedModule, ExpectedClass, ExpectedCallbackFunction),
+                        NativeFrame,
+                        FormatFrame(ExpectedModule, ExpectedClass, ExpectedFunction),
+                    };
+
+                    var actualFrames = new List<string>();
+
+                    while ((line = reader.ReadLine()) != null)
+                    {
+                        line = line.TrimStart();
+                        if (actualFrames.Count == 3)
```

Comment:
Would it make sense here to make sure that `actualFrames` doesn't have extra frames? I.e. the first three frames of actual/expected are the same, but since we break out of the loop we never see if additional actual frames were going to be added that shouldn't be there.

Reply:
I want to test for a run of frames I know about. I can't really guarantee that there won't be extra frames beyond that, especially since I am relying on a wait handle to stop the app at a particular point.